odoo/models.py

```diff
@@ -2866,6 +2866,8 @@ def check_field_access_rights(self, operation, fields):
         def valid(fname):
             """ determine whether user has access to field ``fname`` """
             field = self._fields.get(fname)
+            if not field.exportable:
+                return None
             if field and field.groups:
```

Comment:
The logic is not correct.  If a field has groups the user does not belong to, it must return `False`.
The code should be something like:
```py
if field:
    if field.groups and not self.user_has_groups(field.groups):
        return False
    if not field.exportable:
        return None
return True
```

Reply:
done